components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt

```diff
@@ -191,13 +196,27 @@ class AddonsManagerAdapter(
             }
         }
 
-        scope.launch {
-            val iconBitmap = addonCollectionProvider.getAddonIconBitmap(addon)
+        fetchIcon(addon, holder.iconView)
+    }
 
-            iconBitmap?.let {
-                MainScope().launch {
-                    holder.iconView.setImageBitmap(it)
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    @Suppress("TooGenericExceptionCaught")
```

Comment:
Unfortunately we are getting an `org.mozilla.geckoview.WebRequestError`, for this reason we have to catch it as `Exception` as we don't want to have any `GeckoView` decencies on this component.

Reply:
Because we do not know about this type? I think this type leaking is a problem that we should fix a level higher. Is this coming out of the concept-fetch implementation? This should never throw anything that is not in `concept-fetch` or the standard library (e.g. `IOException`).

Reply:
Thanks for catching this 👍 , I double-checked  we are already wrapping the exception, in [GeckoViewFetchClient](https://github.com/mozilla-mobile/android-components/blob/11b80fbfc662e6100a0d11dafc9264f85d2b7357/components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/fetch/GeckoViewFetchClient.kt#L62), It was my mistake I got confused I saw in the logs `org.mozilla.geckoview.WebRequestError ` but the exception was already wrapped in an `IOException`. 

Updating the pr!

Reply:
Ah, great. :) 👍